hubbard/plot/plot.py

```diff
@@ -249,23 +249,24 @@ def real_space_grid(v, grid_unit, density):
             # Create the real-space grid
             grid = sisl.Grid(grid_unit, sc=g.sc, geometry=g)
 
-            if density:
-                D = sisl.physics.DensityMatrix(g)
-                a = np.arange(len(D))
-                D.D[a, a] = v
-                D.density(grid)
-            else:
+            if mode in ['wavefunction', 'density']:
```

Comment:
this is where it gets blury. I would stick with `mode in ('wavefunction')`. The function `wavefunction` does _not_ give any density, so they shouldn't be confused.